torchtune/modules/kv_cache.py

```diff
@@ -64,7 +65,7 @@ def update(
             Tuple[torch.Tensor, torch.Tensor]: Updated KV cache with key first
         """
         assert input_pos.shape[0] == k_val.shape[2]
-        self.size = input_pos.max().item() + 1
+        self.size = input_pos.max() + 1
```

Comment:
Is this just to get rid of a sync? Or was this actually broken before

Reply:
Just to get rid of the sync and match what's being done in TransformerDecoder